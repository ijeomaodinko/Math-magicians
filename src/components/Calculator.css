* {
  box-sizing: border-box;
  font-family: Verdana, Tahoma, sans-serif;
}

body {
  height: auto;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: aliceblue;
  margin-top: 6.6rem;
  margin-bottom: 0;
}

.wrapper {
  width: 100%;
  background-color: #e9eaeb;
}

button {
  background-color: #e2e2eb;
  border: none;
  color: black;
  padding: 15px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 19px;
  margin: 1px 1px;
  cursor: pointer;
  min-height: 30px;
  min-width: 81px;
  margin-top: 0;
}

button.zero {
  padding-left: 20px;
  padding-right: 20px;
  min-width: 164.5px;
}

button.last {
  background-color: #fbb034;
}

button:hover {
  box-shadow: 0 12px 16px 0 rgba(0, 0, 0, 0.24), 0 17px 50px 0 rgba(0, 0, 0, 0.19);
}

button:active {
  background-color: rgb(204, 203, 207);
}

.screen {
  min-width: 100%;
  min-height: 4rem;
  background-color: #4357692d;
  margin-top: 0;
  margin-bottom: 0;
}

div.screen {
  text-align: right;
}

.input {
  text-align: right;
  font-size: 29px;
  margin-right: 6px;
  color: white;
}

p.input {
  padding-top: 1rem;
}
