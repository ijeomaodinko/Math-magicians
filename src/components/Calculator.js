import React from 'react';
import calculate from '../logic/calculate';

// eslint-disable-next-line react/prefer-stateless-function
class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
    this.clickHandle = this.clickHandle.bind(this);
  }

  clickHandle = (event) => {
    const calc = calculate(this.state, event.target.textContent);
    this.setState(calc);
  };

  render() {
    const {
      total,
      next,
      operation,
    } = this.state;

    return (
      <div className="imenu">
        <div>
          <h2 className="menup">Let&apos;s do some Math</h2>
        </div>
        <div className="wrapper">
          <div className="screen">
            <span className="input">{total}</span>
            <span className="input">{operation}</span>
            <span className="input">{next}</span>
          </div>
          <div className="button-box">
            <button type="button" onClick={this.clickHandle} className="allClear">AC</button>
            <button type="button" onClick={this.clickHandle} className="plusMinus">+/-</button>
            <button type="button" onClick={this.clickHandle} className="percent">%</button>
            <button type="button" onClick={this.clickHandle} className="divide last">รท</button>
            <br />

            <button type="button" onClick={this.clickHandle} className="seven">7</button>
            <button type="button" onClick={this.clickHandle} className="eight">8</button>
            <button type="button" onClick={this.clickHandle} className="nine">9</button>
            <button type="button" onClick={this.clickHandle} className="multiply last">x</button>
            <br />

            <button type="button" onClick={this.clickHandle} className="four">4</button>
            <button type="button" onClick={this.clickHandle} className="five">5</button>
            <button type="button" onClick={this.clickHandle} className="six">6</button>
            <button type="button" onClick={this.clickHandle} className="subtract last">-</button>
            <br />

            <button type="button" onClick={this.clickHandle} className="one">1</button>
            <button type="button" onClick={this.clickHandle} className="two">2</button>
            <button type="button" onClick={this.clickHandle} className="three">3</button>
            <button type="button" onClick={this.clickHandle} className="add  last">+</button>
            <br />

            <button type="button" onClick={this.clickHandle} className="zero">0</button>
            <button type="button" onClick={this.clickHandle} className="point">.</button>
            <button type="button" onClick={this.clickHandle} className="equalto  last">=</button>
          </div>
        </div>
      </div>
    );
  }
}

export default Calculator;
