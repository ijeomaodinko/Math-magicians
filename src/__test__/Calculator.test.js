import { render, screen } from '@testing-library/react';
import Calculator from '../components/Calculator';
import '@testing-library/jest-dom';

test('Display text on the screen', () => {
  render(<Calculator />);
  const element = screen.getByText(/Let's do some Math/i);
  expect(element).toBeInTheDocument();
});

it('matches snapshot', () => {
  const tree = render(<Calculator />);
  expect(tree).toMatchSnapshot();
});
