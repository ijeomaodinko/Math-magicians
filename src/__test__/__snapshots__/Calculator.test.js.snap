// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="imenu"
      >
        <div>
          <h2
            class="menup"
          >
            Let's do some Math
          </h2>
        </div>
        <div
          class="wrapper"
        >
          <div
            class="screen"
          >
            <span
              class="input"
            />
            <span
              class="input"
            />
            <span
              class="input"
            />
          </div>
          <div
            class="button-box"
          >
            <button
              class="allClear"
              type="button"
            >
              AC
            </button>
            <button
              class="plusMinus"
              type="button"
            >
              +/-
            </button>
            <button
              class="percent"
              type="button"
            >
              %
            </button>
            <button
              class="divide last"
              type="button"
            >
              รท
            </button>
            <br />
            <button
              class="seven"
              type="button"
            >
              7
            </button>
            <button
              class="eight"
              type="button"
            >
              8
            </button>
            <button
              class="nine"
              type="button"
            >
              9
            </button>
            <button
              class="multiply last"
              type="button"
            >
              x
            </button>
            <br />
            <button
              class="four"
              type="button"
            >
              4
            </button>
            <button
              class="five"
              type="button"
            >
              5
            </button>
            <button
              class="six"
              type="button"
            >
              6
            </button>
            <button
              class="subtract last"
              type="button"
            >
              -
            </button>
            <br />
            <button
              class="one"
              type="button"
            >
              1
            </button>
            <button
              class="two"
              type="button"
            >
              2
            </button>
            <button
              class="three"
              type="button"
            >
              3
            </button>
            <button
              class="add  last"
              type="button"
            >
              +
            </button>
            <br />
            <button
              class="zero"
              type="button"
            >
              0
            </button>
            <button
              class="point"
              type="button"
            >
              .
            </button>
            <button
              class="equalto  last"
              type="button"
            >
              =
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="imenu"
    >
      <div>
        <h2
          class="menup"
        >
          Let's do some Math
        </h2>
      </div>
      <div
        class="wrapper"
      >
        <div
          class="screen"
        >
          <span
            class="input"
          />
          <span
            class="input"
          />
          <span
            class="input"
          />
        </div>
        <div
          class="button-box"
        >
          <button
            class="allClear"
            type="button"
          >
            AC
          </button>
          <button
            class="plusMinus"
            type="button"
          >
            +/-
          </button>
          <button
            class="percent"
            type="button"
          >
            %
          </button>
          <button
            class="divide last"
            type="button"
          >
            รท
          </button>
          <br />
          <button
            class="seven"
            type="button"
          >
            7
          </button>
          <button
            class="eight"
            type="button"
          >
            8
          </button>
          <button
            class="nine"
            type="button"
          >
            9
          </button>
          <button
            class="multiply last"
            type="button"
          >
            x
          </button>
          <br />
          <button
            class="four"
            type="button"
          >
            4
          </button>
          <button
            class="five"
            type="button"
          >
            5
          </button>
          <button
            class="six"
            type="button"
          >
            6
          </button>
          <button
            class="subtract last"
            type="button"
          >
            -
          </button>
          <br />
          <button
            class="one"
            type="button"
          >
            1
          </button>
          <button
            class="two"
            type="button"
          >
            2
          </button>
          <button
            class="three"
            type="button"
          >
            3
          </button>
          <button
            class="add  last"
            type="button"
          >
            +
          </button>
          <br />
          <button
            class="zero"
            type="button"
          >
            0
          </button>
          <button
            class="point"
            type="button"
          >
            .
          </button>
          <button
            class="equalto  last"
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
